homeassistant/components/stream/hls.py

```diff
@@ -68,24 +68,23 @@ class HlsPlaylistView(StreamView):
     cors_allowed = True
 
     @staticmethod
-    def render_preamble(track):
-        """Render preamble."""
-        return [
-            "#EXT-X-VERSION:6",
-            f"#EXT-X-TARGETDURATION:{track.target_duration}",
-            '#EXT-X-MAP:URI="init.mp4"',
-        ]
-
-    @staticmethod
-    def render_playlist(track):
+    def render(track):
         """Render playlist."""
-        segments = list(track.get_segments())[-NUM_PLAYLIST_SEGMENTS:]
+        # NUM_PLAYLIST_SEGMENTS+1 because most recent is probably not yet complete
+        segments = list(track.get_segments())[-(NUM_PLAYLIST_SEGMENTS + 1) :]
 
-        if not segments:
-            return []
+        # To cap the number of complete segments at NUM_PLAYLIST_SEGMENTS,
+        # remove the first segment if the last segment is actually complete
+        if segments[-1].complete:
+            segments = segments[-NUM_PLAYLIST_SEGMENTS:]
 
         first_segment = segments[0]
         playlist = [
+            "#EXTM3U",
+            "#EXT-X-VERSION:6",
+            "#EXT-X-INDEPENDENT-SEGMENTS",
```

Comment:
Makes sense to mark EXT-X-INDEPENDENT-SEGMENTS as our segments all start with a video keyframe and thus can be played independently. See https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-4.4.2.1